recurring_contract/tests/test_base_contract.py

```diff
@@ -0,0 +1,122 @@
+            ('user_type', '=', account_type)]).ids[0]
+        account_type = account_type_obj.search([
+            ('code', '=', 'payable')]).ids[0]
+        property_account_payable = account_obj.search([
+            ('type', '=', 'payable'),
+            ('user_type', '=', account_type)]).ids[0]
+
+        # Creation of partners
+        partner_obj = self.env['res.partner']
+        self.partners = partner_obj.create(
+            {
+                'name': 'Monsieur Client 137',
+                'property_account_receivable': property_account_receivable,
+                'property_account_payable': property_account_payable,
+                'notify_email': 'none',
+                'ref': '00001111',
+            })
+        self.partners += partner_obj.create(
+            {
+                'name': 'Monsieur Pumba',
+                'property_account_receivable': property_account_receivable,
+                'property_account_payable': property_account_payable,
+                'notify_email': 'none',
+                'ref': '00002222',
+            })
+        self.partners += partner_obj.create({
+            'name': 'Monsieur Bryan',
+            'property_account_receivable': property_account_receivable,
+            'property_account_payable': property_account_payable,
+            'notify_email': 'always',
+            'ref': '00003333',
+        })
+        # Creation of payement terms
+        payment_term_obj = self.env['account.payment.term']
+        self.payment_term_id = payment_term_obj.search(
+            [('name', '=', '15 Days')])[0].id
+
+    def _create_contract(self, start_date, group, next_invoice_date,
+                         other_vals=None):
+        """
+            Create a contract. For that purpose we have created a partner
+            to get his id
+        """
+        # Creation of a contract
+        contract_obj = self.env['recurring.contract']
+        if other_vals and 'type' in other_vals:
+            contract_obj = contract_obj.with_context(
+                default_type=other_vals['type'])
+        partner_id = group.partner_id.id
+        vals = {
+            'start_date': start_date,
+            'partner_id': partner_id,
+            'group_id': group.id,
+            'next_invoice_date': next_invoice_date,
+        }
+        if other_vals:
+            vals.update(other_vals)
+        contract = contract_obj.create(vals)
+        return contract
+
+    def _create_contract_line(self, contract_id, price, other_vals=None):
+        """ Create contract's lines """
+        contract_line_obj = self.env['recurring.contract.line']
+        vals = {
+            'product_id': 1,
+            'amount': price,
+            'contract_id': contract_id,
+        }
+        if other_vals:
+            vals.update(other_vals)
+
+        contract_line = contract_line_obj.create(vals)
+        return contract_line
+
+    def _create_group(self, change_method, partner_id,

```

Comment:
Put create_group method at first of private methods, because logic seems to be to create a group first, then a contract.